src/libstd/sys/unix/rwlock.rs

```diff
@@ -9,16 +9,22 @@
 // except according to those terms.
 
 use libc;
-use cell::UnsafeCell;
+use cell::{Cell, UnsafeCell};
 
-pub struct RWLock { inner: UnsafeCell<libc::pthread_rwlock_t> }
+pub struct RWLock {
+    inner: UnsafeCell<libc::pthread_rwlock_t>,
+    write_locked: Cell<bool>,
```

Comment:
Semantically the usage here is fine, but `Cell` is not `Sync` and it's technically used in an unsafe fashion here. It's not normally safe to access it on many threads, but the readers will read it concurrently here.

But yeah this would just to be a bit "more correct", it wouldn't actually change anything about the implementation.